src/NuGetGallery/Controllers/UsersController.cs

```
@@ -462,62 +463,128 @@ public virtual ActionResult Thanks()
+                OwnerRequests = GetOwnerRequestsForUser(currentUser),
+                ReservedNamespaces = GetReservedNamespacesForUser(currentUser),
+                WasMultiFactorAuthenticated = User.WasMultiFactorAuthenticated(),
+                IsCertificatesUIEnabled = ContentObjectService.CertificatesConfiguration?.IsUIEnabledForUser(currentUser) ?? false
+            };
 
-            var owners = new List<ListPackageOwnerViewModel> {
-                new ListPackageOwnerViewModel
+            return View(model);
+        }
+
+        [HttpGet]
+        [UIAuthorize]
+        public virtual JsonResult PackagesPaged(bool listed, int page = 0, string username = null)
+        {
+            return Json(
+                GetPagedPackagesForManagePackages(listed, page, username),
+                JsonRequestBehavior.AllowGet);
+        }
+
+        private dynamic GetPagedPackagesForManagePackages(bool listed, int page = 0, string username = null)
+        {
+            var currentUser = GetCurrentUser();
+            IEnumerable<Package> packages;
+            if (username == null)
+            {
+                packages = PackageService.FindPackagesByAnyMatchingOwner(
+                    currentUser, includeUnlisted: true, includeVersions: true);
+            }
+            else
+            {
+                var user = UserService.FindByUsername(username);
+                if (user != null)
+                {
+                    packages = PackageService.FindPackagesByOwner(
+                        user, includeUnlisted: true, includeVersions: true);
+                }
+                else
                 {
-                    Username = "All packages"
-                },
-                new ListPackageOwnerViewModel(currentUser)
-            }.Concat(currentUser.Organizations.Select(o => new ListPackageOwnerViewModel(o.Organization)));
+                    packages = Enumerable.Empty<Package>();
```

This could indicate a bug, or at least stale data... should we throw instead? Probably should match behavior of whatever we do if current user doesn't have permissions.

---

In general, I decided not to implement any stale data handling for the paging because it would complicate things substantially and provide little benefit. Not only is a users' list of packages and owners unlikely to change, but there isn't any sort of "e-tag" or other identifier for the client to determine that its cache is out of date.

Technically, I could make this return a `400` or `404`, but then I would have to add handling for this case on the client-side. By instead returning an empty list, I don't have to do any additional work--the client side already handles empty lists. I think, with this logic, it may be a problem if a user loads the first page of results for a user, but then that user is deleted..but again, this is extraordinarily unlikely.

---

Though about it again--in cases that the page has a bug, I think we should return an error code so we should display to users that the page is having issues. I think the stale data problems are less of a concern though.